rasa/core/policies/policy.py

```diff
@@ -354,49 +381,50 @@ def _metadata_filename(cls) -> Text:
         """
         pass
 
-    def persist(self, path: Union[Text, Path]) -> None:
-        """Persists the policy to storage.
-
-        Args:
-            path: Path to persist policy to.
-        """
-        # not all policies have a featurizer
-        if self.featurizer is not None:
-            self.featurizer.persist(path)
+    def persist(self) -> None:
+        """Persists the policy to storage."""
+        with self._model_storage.write_to(self._resource) as path:
+            # not all policies have a featurizer
+            if self.featurizer is not None:
+                self.featurizer.persist(path)
 
-        file = Path(path) / self._metadata_filename()
+            file = Path(path) / self._metadata_filename()
 
-        rasa.shared.utils.io.create_directory_for_file(file)
-        rasa.shared.utils.io.dump_obj_as_json_to_file(file, self._metadata())
+            rasa.shared.utils.io.create_directory_for_file(file)
+            rasa.shared.utils.io.dump_obj_as_json_to_file(file, self._metadata())
 
     @classmethod
-    def load(cls, path: Union[Text, Path], **kwargs: Any) -> "Policy":
-        """Loads a policy from path.
-
-        Args:
-            path: Path to load policy from.
-
-        Returns:
-            An instance of `Policy`.
-        """
-        metadata_file = Path(path) / cls._metadata_filename()
+    def load(
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+        **kwargs: Any,
+    ) -> "PolicyGraphComponent":
+        """Loads a trained policy (see parent class for full docstring)."""
+        config = {}
+        featurizer = None
 
-        if metadata_file.is_file():
-            data = json.loads(rasa.shared.utils.io.read_file(metadata_file))
+        try:
+            with model_storage.read_from(resource) as path:
+                metadata_file = Path(path) / cls._metadata_filename()
+                config = json.loads(rasa.shared.utils.io.read_file(metadata_file))
 
-            if (Path(path) / FEATURIZER_FILE).is_file():
-                featurizer = TrackerFeaturizer.load(path)
-                data["featurizer"] = featurizer
+                if (Path(path) / FEATURIZER_FILE).is_file():
+                    featurizer = TrackerFeaturizer.load(path)
 
-            data.update(kwargs)
+                config.update(kwargs)
 
-            return cls(**data)
+        except (ValueError, FileNotFoundError, FileIOException):
+            logger.info(
+                f"Couldn't load metadata for policy '{cls.__name__}' as the persisted "
+                f"metadata couldn't be loaded."
+            )
```

Comment:
I just did what we did before as I didn't want to refactor but merely adapt the policies 🤷🏻‍♂️ I think someone probably put `info` in case this happens because the policy didn't receive the necessary training data to train itself. In that case `error`  during prediction time could be misleading

Reply:
Oh ok, interesting - Yeah there are lots of cases where training is silently skipped :/